StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1111ClosingParenthesisMustBeOnLineOfLastParameter.cs

```diff
@@ -54,7 +58,154 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+                objectCreation.ArgumentList.IsMissing ||
+                !objectCreation.ArgumentList.Arguments.Any())
+            {
+                return;
+            }
+
+            var lastArgument = objectCreation.ArgumentList
+                .Arguments
+                .Last();
+
+            if (!objectCreation.ArgumentList.CloseParenToken.IsMissing &&
+                !lastArgument.IsMissing)
+            {
+                CheckIfLocationOfLastArgumentOrParameterAndCloseTokenAreTheSame(context, lastArgument,
+                    objectCreation.ArgumentList.CloseParenToken);
+            }
+
+        }
+
+
+        private void HandleIndexerDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var indexerDeclaration = (IndexerDeclarationSyntax)context.Node;
+
+            if (indexerDeclaration.ParameterList == null ||
+                !indexerDeclaration.ParameterList.IsMissing &&
+                !indexerDeclaration.ParameterList.Parameters.Any())
+            {
+                return;
+            }
+
+            var lastParameter = indexerDeclaration.ParameterList
+                         .Parameters
+                         .Last();
+
+            if (!indexerDeclaration.ParameterList.CloseBracketToken.IsMissing)
+            {
+                CheckIfLocationOfLastArgumentOrParameterAndCloseTokenAreTheSame(context, lastParameter, indexerDeclaration.ParameterList.CloseBracketToken);
+            }
+        }
+
+        private static void HandleBaseMethodDeclaration(SyntaxNodeAnalysisContext context,
+            BaseMethodDeclarationSyntax baseMethodDeclarationSyntax)
+        {
+            if (baseMethodDeclarationSyntax.ParameterList == null ||
+                baseMethodDeclarationSyntax.ParameterList.IsMissing ||
+                !baseMethodDeclarationSyntax.ParameterList.Parameters.Any())
+            {
+                return;
+            }
+
+            var lastParameter = baseMethodDeclarationSyntax.ParameterList
+                .Parameters
+                .Last();
+
+            if (!baseMethodDeclarationSyntax.ParameterList.CloseParenToken.IsMissing)
+            {
+                CheckIfLocationOfLastArgumentOrParameterAndCloseTokenAreTheSame(context, lastParameter, baseMethodDeclarationSyntax.ParameterList.CloseParenToken);
+            }
+        }
+
+        private static void CheckIfLocationOfLastArgumentOrParameterAndCloseTokenAreTheSame(SyntaxNodeAnalysisContext context,
+       CSharpSyntaxNode parameterOrArgument, SyntaxToken closeToken)
+        {
+            var lastParameterLine = parameterOrArgument.GetLocation().GetLineSpan();
+            var closeParenLocation = closeToken.GetLocation();
+            var closeParenLine = closeParenLocation.GetLineSpan();
+            if (lastParameterLine.IsValid &&
+                closeParenLine.IsValid &&
+                closeParenLine.StartLinePosition.Line != lastParameterLine.StartLinePosition.Line)
```

Comment:
I wouldn't report this. I read this as the closing parenthesis must be on the same line as the *last character* of the last argument.